hugr-model/src/v0/text/parse.rs

```diff
@@ -50,33 +52,59 @@ pub fn parse<'a>(input: &'a str, bump: &'a Bump) -> Result<ParsedModule<'a>, Par
 struct ParseContext<'a> {
     module: Module<'a>,
     bump: &'a Bump,
+    vars: VarTable<'a>,
+    links: LinkTable<&'a str>,
+    symbols: SymbolTable<'a>,
+    implicit_imports: FxHashMap<&'a str, NodeId>,
 }
 
 impl<'a> ParseContext<'a> {
     fn new(bump: &'a Bump) -> Self {
         Self {
             module: Module::default(),
+            symbols: SymbolTable::default(),
+            links: LinkTable::default(),
+            vars: VarTable::default(),
+            implicit_imports: FxHashMap::default(),
             bump,
         }
     }
 
     fn parse_module(&mut self, pair: Pair<'a, Rule>) -> ParseResult<()> {
         debug_assert_eq!(pair.as_rule(), Rule::module);
         let mut inner = pair.into_inner();
+
+        self.module.root = self.module.insert_region(Region::default());
+        self.symbols.enter(self.module.root);
+        self.links.enter(self.module.root);
+
+        // TODO: What scope does the metadata live in?
```

Comment:
I'd guess that metadata attached to a node lives just outside the scope of that node itself (e.g. cannot refer to variables defined by that node, or stuff inside it), but can refer to anything outside it. That's a bit of a guess, though, happy to leave this as TODO.

Reply:
Yes, this is the indended scoping for metadata and it is implemented as such for almost all nodes. However this is the root node: everything, _including imports_, lives within the root node. So the metadata on the root node can not refer to anything...